lib/galaxy/datatypes/hubAssembly.py

```diff
@@ -0,0 +1,59 @@
+"""
+HubAssembly datatype
+"""
+import logging
+import galaxy.version as version
+
+# Support for Galaxy <= 16.01
+if version.VERSION_MAJOR <= "16.01":
+    from galaxy.datatypes.images import Html
+else:
+    from galaxy.datatypes.text import Html
+
+log = logging.getLogger(__name__)
+
+
+class HubAssembly( Html ):
+    """
+    derived class for BioC data structures in Galaxy
+    """
+
+    file_ext = 'huba'
+    composite_type = 'auto_primary_file'
+
+    def __init__(self, **kwd):
+        Html.__init__(self, **kwd)
+
+    def generate_primary_file( self, dataset=None ):
+        """
+        This is called only at upload to write the html file
+        cannot rename the datasets here - they come with the default unfortunately
+        """
+        rval = [
+            '<html><head><title>Files for Composite Dataset (%s)</title></head><p/>\
+            This composite dataset is composed of the following files:<p/><ul>' % (
+                self.file_ext)]
+        for composite_name, composite_file in self.get_composite_files( dataset=dataset ).iteritems():
```

Comment:
Ahh, I see, you were thinking about python3.  Good to have the rationale here instead of simply 'just items()' :)

Definitely not a requirement to change this now @remimarenco, up to you.